app/validation/validator.py

```diff
@@ -2064,24 +2160,55 @@ def _get_answers_with_parent_ids(self, json_to_validate):
 
         return answers
 
-    def _get_questions_with_context(self, questionnaire_json):
+    @classmethod
+    def _get_option_value_to_answer_id_map(cls, json_to_validate):
+        option_value_to_answer_id_map = {}
+        questions = (
+            question
+            for question, _ in cls._get_questions_with_context(json_to_validate)
+        )
+        radio_and_checkbox_answers = cls._get_answers_by_type_from_questions(
+            questions=questions, answer_types={"Radio", "Checkbox"}
+        )
+
+        for answer in radio_and_checkbox_answers:
+            answer_id = answer["id"]
+            option_values = [option["value"] for option in answer["options"]]
+
+            if answer_id in option_value_to_answer_id_map:
+                option_value_to_answer_id_map[answer_id].update(option_values)
+            else:
+                option_value_to_answer_id_map[answer_id] = set(option_values)
```

Comment:
If you change `option_value_to_answer_id_map = {}` to `option_value_to_answer_id_map = defaultdict(set)` then these four lines can be simplified to `option_value_to_answer_id_map[answer_id].update(option_values)`.

Reply:
Nice one.